packages/vite/src/node/plugins/assetImportMetaUrl.ts

```diff
@@ -65,6 +71,16 @@ export function assetImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
                 continue
               }
 
+              const query = getQuery(queryString)
+              const globOptions = {
+                eager: true,
+                import: 'default',
+                query: {
+                  // A hack to allow 'as' & 'query' exist at the same time
+                  url: '',
+                  ...query,
+                },
```

Comment:
I checked if we should still use `as` here but it isn't possible to use both at the same time. We could still use `as` if there is no `query` but that adds more complexity. I think going for query all the time is good here (we may also remove `as` in the future now that the proposal evolved, again)

Reply:
I'd prefer if we avoid `ufo`'s `getQuery` since it uses a lot of regexes:

https://github.com/unjs/ufo/blob/a809630deb75ac20edb93f1612b50d9a477297f9/src/parse.ts#L29-L52

https://github.com/unjs/ufo/blob/a809630deb75ac20edb93f1612b50d9a477297f9/src/query.ts#L16-L42

We can use `injectQuery` instead to keep everything as a string, since `importMetaGlob` would also convert the object as a string.